### ISSUE TITLE
TDDFT triplet excited state geometry optimization

### ISSUE THREAD

**Issue by ghost** [pyscf/pyscf#1728]
Hello,

I am new at using pySCF and would like to perform a geometry optimization on a T1 excited state.

I tried using the following Code with mf being my RKS result:
```
mytd = tddft.TDA(mf)
mytd.nstates = 10
mytd.singlet = False

excited_grad = mytd.nuc_grad_method().as_scanner(state=1)
mol1 = geomopt.optimize(excited_grad)
```

Unfortunately, this gave the following error:
NotImplementedError: GGA triplet

Can someone explain me, what the issue is?

**Comment by matthew-hennefarth** [comment]
May I ask which KS-DFT Functional you are using?

**Comment by missymetrum** [comment]
`from pyscf import gto, dft, scf, tddft
from pyscf import tdscf
from pyscf.geomopt.geometric_solver import optimize
from pyscf import geomopt

mol = gto.Mole()
mol.atom = ''' `

.
.
.

`'''
mol.basis = '631g*'
mol.verbose = 4
mol.build()

mf = dft.RKS(mol)
mf.xc = 'CAMB3LYP'
mf.chkfile = 'hf.chk'  # Store SCF information to enable restart
mf.kernel()

mytd = tddft.TDA(mf)
mytd.nstates = 10
mytd.singlet = False

excited_grad = mytd.nuc_grad_method().as_scanner(state=1)
mol1 = geomopt.optimize(excited_grad)

print('\n \n \n Finale Geometry: \n ---------------')
print(mol1.atom_coords(unit='Angstrom'))
print(mol1._atom)`

**Comment by xubwa** [comment]
As the name of the error suggests, the gradient for triplet state is not implemented. In your case, if you just want to optimize the triplet ground state, then maybe you could try just use a uks or a roks reference without the need to do tddft.

**Comment by verena-neufeld** [comment]
Hi @missymetrum , thanks for posting the issue here. Has it been answered?

**Comment by sunqm** [comment]
Related to PR #1714. This feature is available in pyscf-2.3 and newer.
